clients/src/main/java/org/apache/kafka/clients/consumer/internals/AsyncKafkaConsumer.java

```diff
@@ -1132,7 +1135,11 @@ private Map<TopicPartition, Long> beginningOrEndOffset(Collection<TopicPartition
 
             Map<TopicPartition, OffsetAndTimestampInternal> offsetAndTimestampMap;
             try {
-                offsetAndTimestampMap = applicationEventHandler.addAndGet(listOffsetsEvent);
+                applicationEventHandler.add(listOffsetsEvent);
+                offsetAndTimestampMap = processBackgroundEvents(
+                        listOffsetsEvent.future(),
+                        timer, __ -> false
+                );
```

Comment:
As I understand it, we need to check for the errors from the background thread. But do we need to check _repeatedly_ during the execution of the `ListOffsetsEvent`, or can we just check once beforehand?

```suggestion
                processBackgroundEvents();
                offsetAndTimestampMap = applicationEventHandler.addAndGet(listOffsetsEvent);
```

Reply:
> Is it sufficient to perform a single check of the background events before submitting the application event, or do we really need to perform multiple checks of the background events while we wait for the application event to complete?

It can't process ` processBackgroundEvents()` only once before `applicationEventHandler.addAndGet`. Test will be fail if I change to below, I think a loop for processBackgroundEvents is necessary
```java
 processBackgroundEvents();
offsetAndTimestampMap = applicationEventHandler.addAndGet(listOffsetsEvent);
```